embassy-stm32/src/dma/dma_bdma.rs

```diff
@@ -828,7 +828,6 @@ impl<'a, W: Word> ReadableRingBuffer<'a, W> {
     /// You must call this after creating it for it to work.
     pub fn start(&mut self) {
         self.channel.start();
-        self.clear();
```

Comment:
Clear on start conflicts with `write_immediate`. With `clear` in place, `write_immediate` would have no effect.